packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/index.js

```diff
@@ -59,7 +61,20 @@ const LocaleSettingsPage = () => {
         title={listTitle}
         items={locales}
         isLoading={isLoading}
-        customRowComponent={locale => <LocaleRow locale={locale} />}
+        customRowComponent={locale => <LocaleRow locale={locale} onDelete={handleDeleteLocale} />}
+      />
+
+      <PopUpWarning
```

Comment:
Can you use the `ModalConfirm` component instead and implement the basic logique to handle the deletion?


```
const handleConfirm = useCallback(() => {
    setShowButtonLoader(true);

    setTimeout(() => {
      setShowModal(false);
    }, 1000);
  }, []);

  const handleClosed = useCallback(() => {
    setLocaleToDelete(null);
  }, []);


<ModalConfirm
        isOpen={showModal}
        toggle={handleToggle}
        type="xwarning"
        onConfirm={handleConfirm}
        onClosed={handleClosed}
        showButtonLoader={showButtonLoader}
        confirmButtonLabel={{
          id: getTrad('Settings.locales.modal.delete.confirm'),
        }}
        content={{
          id: getTrad('Settings.locales.modal.delete.message'),
        }}
      >
        <Text fontWeight="bold">
          {formatMessage({ id: getTrad('Settings.locales.modal.delete.secondMessage') })}
        </Text>
      </ModalConfirm>
```

Reply:
I think it will also be worth to use a reducer instead of a local state.